packages/manifold/contracts/edition/ManifoldERC721Edition.sol

```diff
@@ -29,101 +29,102 @@ contract ManifoldERC721Edition is CreatorExtension, ICreatorExtensionTokenURI, I
+        if (_maxSupply[creatorCore][instanceId] == 0) revert("Invalid instanceId");
+        return _maxSupply[creatorCore][instanceId];
     }
 
     /**
      * See {IManifoldERC721Edition-setTokenURIPrefix}.
      */
-    function setTokenURIPrefix(address creator, uint256 series, string calldata prefix) external override creatorAdminRequired(creator) {
-        require(series > 0 && series <= _currentSeries[creator], "Invalid series");
-        _tokenPrefix[creator][series] = prefix;
+    function setTokenURIPrefix(address creatorCore, uint256 instanceId, string calldata prefix) external override creatorAdminRequired(creatorCore) {
+        if (_maxSupply[creatorCore][instanceId] == 0) revert("Invalid instanceId");
+        _tokenPrefix[creatorCore][instanceId] = prefix;
     }
     
     /**
      * @dev See {ICreatorExtensionTokenURI-tokenURI}.
      */
-    function tokenURI(address creator, uint256 tokenId) external view override returns (string memory) {
-        (uint256 series, uint256 index) = _tokenSeriesAndIndex(creator, tokenId);
-        return string(abi.encodePacked(_tokenPrefix[creator][series], (index+1).toString()));
+    function tokenURI(address creatorCore, uint256 tokenId) external view override returns (string memory) {
+        (uint256 instanceId, uint256 index) = _tokenInstanceAndIndex(creatorCore, tokenId);
+        // which version of core
+        
+        return string(abi.encodePacked(_tokenPrefix[creatorCore][instanceId], (index+1).toString()));
     }
     
     /**
      * @dev See {IManifoldERC721Edition-mint}.
      */
-    function mint(address creator, uint256 series, address recipient, uint16 count) external override nonReentrant creatorAdminRequired(creator) {
-        require(count > 0, "Invalid amount requested");
-        require(_totalSupply[creator][series]+count <= _maxSupply[creator][series], "Too many requested");
-        
-        uint256[] memory tokenIds = IERC721CreatorCore(creator).mintExtensionBatch(recipient, count);
-        _updateIndexRanges(creator, series, tokenIds[0], count);
+    function mint(address creatorCore, uint256 instanceId, uint256 currentSupply, IManifoldERC721Edition.Recipient[] memory recipients) external override nonReentrant creatorAdminRequired(creatorCore) {        
+        if (_totalSupply[creatorCore][instanceId] != currentSupply) revert("Incorrect supply");
+        _mintTokens(creatorCore, recipients, instanceId);
+    }
+
+    function _mintTokens(address creatorCore, Recipient[] memory recipients, uint256 instanceId) private {
+        if (_totalSupply[creatorCore][instanceId]+1 > _maxSupply[creatorCore][instanceId]) revert("Too many requested");
+        if (recipients.length == 0) revert("No recipients");
+
+        // Grab the start index by minting off a first token...
+        uint256 startIndex = IERC721CreatorCore(creatorCore).mintExtension(recipients[0].recipient);
```

Comment:
This is wrong though.  you need to call mintExtensionBatch with the count of the first recipient.  Also you're double minting for the first recipient.

I'm pretty sure mintExtensionBatch will revert with count 0.

The logic is simpler now isn't it?

    uint256[] memory tokenIdResults;
    for (uint256 i; i < recipients.length;) {
           if (count == 0) revert InvalidCount();
            count += recipients[i].count;
            if (_totalSupply[creatorCore][instanceId]+count > _maxSupply[creatorCore][instanceId]) revert TooManyRequested();
            tokenIdResults = IERC721CreatorCore(creatorCore).mintExtensionBatch(recipients[i].recipient, recipients[i].count);
            if (i == 0) startIndex = tokenIdResults[0];
            unchecked{++i;}
        }

Reply:
Ok, updated to use this

Reply:
hold on, I'm making some changes

Reply:
thought of some stuff